java/system-test/system-test-dsl/src/test/java/sleeper/systemtest/dsl/testutil/drivers/InMemoryCompaction.java

```diff
@@ -155,8 +164,16 @@ private void finishTasks() {
 
     private RecordsProcessedSummary compact(CompactionJob job, TableProperties tableProperties, StateStore stateStore, String taskId) {
         Instant startTime = Instant.now();
+        Schema schema = tableProperties.getSchema();
+        PartitionTree partitionTree = null;
+        try {
+            partitionTree = new PartitionTree(stateStore.getAllPartitions());
+        } catch (StateStoreException e) {
+            throw new RuntimeException(e);
+        }
+        Partition partition = partitionTree.getPartition(job.getPartitionId());
```

Comment:
We could extract a method here to get the partition for the job.

Reply:
Done, thanks